dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+    DALI_ENFORCE(sequence_extent_.size() == 1,
+                 make_string("The `", sequenceLayoutArgName, "` must be a single character, got `",
+                             sequence_extent_, "`."));
+  }
+
+  void ProcessInputArgs(const Workspace &ws, int batch_size) {
+    SetupOffsetsAndSizes(ws);
+    shape_.Acquire(spec_, ws, batch_size, ArgValue_EnforceUniform, ArgShapeFromSize<1>{});
+    SetupOutputShape(shape_.get());
+    SetupOutputLayout();
+  }
+
+  auto GetMaxOutChunkVol() const {
+    return max_output_sample_vol_;
+  }
+
+  auto GetTotalChunksNum() const {
+    auto total_chunks_num = sizes_.size();
+    assert(total_chunks_num == offsets_.size());
+    assert(total_chunks_num == chunks_per_sample_.num_elements());
+    return total_chunks_num;
+  }
+
+  const auto &GetChunksNumPerSample() const {
+    return chunks_per_sample_;
+  }
+
+  const TensorListShape<> &GetOutputShape() const {
+    return output_shape_;
+  }
+
+  const TensorLayout &GetOutputLayout() const {
+    return output_layout_;
+  }
+
+  const std::vector<int64_t> &GetInChunkOffsets() const {
+    return offsets_;
+  }
+
+  const std::vector<size_t> &GetInChunkSizes() const {
+    return sizes_;
+  }
+
+  bool HasChunks() const {
+    return HasExplicitChunkOffsets() || HasExplicitChunkSizes();
+  }
+
+ private:
+  bool HasExplicitChunkOffsets() const {
+    return chunk_offsets_.HasExplicitValue();
+  }
+
+  bool HasExplicitChunkSizes() const {
+    return chunk_sizes_.HasExplicitValue();
+  }
+
+  void SetupOutputLayout() {
+    if (layout_.empty()) {
+      return;
+    }
+    auto sample_dim = GetOutputShape().sample_dim() - HasChunks();
+    DALI_ENFORCE(
+        layout_.size() == sample_dim,
+        make_string("The size of inflated chunks' layout must match the number of extents in "
+                    "the output shape. However, got the layout `",
+                    layout_, "` while the chunks output shape has ", sample_dim, " extent(s)."));
+    if (!HasChunks()) {
+      output_layout_ = layout_;
+    } else {
+      assert(sequence_extent_.size() == 1);
+      output_layout_ = sequence_extent_ + layout_;
+    }
+  }
+
+  void SetupOffsetsAndSizes(const Workspace &ws) {
+    const auto &in_shape = ws.GetInputShape(0);
+    int batch_size = in_shape.num_samples();
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      DALI_ENFORCE(in_shape[sample_idx].num_elements() > 0,
+                   make_string("Input sample cannot be empty, but the sample at index ", sample_idx,
+                               " has volume 0."));
+    }
+    if (HasChunks()) {
+      AcquireInferOffsetsSizes(ws);
+    } else {
+      chunks_per_sample_ = uniform_list_shape(batch_size, TensorShape<1>{1});
+      offsets_.resize(batch_size, 0);
+      sizes_.clear();
+      sizes_.reserve(batch_size);
+      for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+        sizes_.push_back(in_shape[sample_idx].num_elements());
+      }
```

Comment:
How about going the other way round and creating a TensorListView from this instead of flattening the data obtained from ArgValue?

Reply:
I though about it, but I do need flattened size_t sizes in the end, so decided on not going over the top with abstraction here and just preprocess the argument to the format that is really needed by the operator.

Reply:
OK.